public_html/script/src/grid+draggable/grid-manager.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+import Grid from './grid';
+
+const [GRIDS, CANVAS_ID] = [Symbol('GRID'), Symbol('CANVAS_ID')];
+
+export const GridManager = class {
+    constructor(canvasID) {
+        this[CANVAS_ID] = canvasID;
+        this[GRIDS] = [];
+    }
+
+    get canvasID() { return this[CANVAS_ID]; }
+
+    addGrid(grid) {
+        if (!(grid instanceof Grid)) {
+            throw new TypeError(
+                'Trying to add a non-grid object to a grid manager',
+                'grid-manager.js'
+            );
+        }
+        let index = this[GRIDS].indexOf(grid);
+        if (index !== -1) {
+            throw new Error(
+                'Trying to add a grid to a grid manager it\'s already in',
+                'grid-manager.js'
+            );
+        }
+        this[GRIDS].push(grid);
+    }
+
+    removeGrid(grid) {
+        if (!(grid instanceof Grid)) {
+            throw new TypeError(
+                'Trying to remove a non-grid object from a grid manager',
+                'grid-manager.js'
+            );
+        }
+        let index = this[GRIDS].indexOf(grid);
+        if (index === -1) {
+            throw new Error(
```

Comment:
I'm gonna stick with throwing; I'd rather have a good message than a cleaner piece of code. Also I don't wanna change them all haha